Created Documentation for issue number 1925 - Valid Brackets Using Stack (#1971)
girlscript/winter-of-contributing: Competitive_Programming/Java/Stacks & Queues/Valid Brackets/Valid-Brackets-Using-Stack.md

<h1 align = "center"><strong>Valid Brackets Using Stack</strong></h1>
<hr>
<h2>Problem Statement :- <br><br><em><strong> • Check for Balanced Brackets in an expression using Stack.</strong></em></h2>
<hr>
<h2><strong>Brackets which we all most commonly use or encounter are of 3 main types :-</strong></h2>
<ul>
    <li>Round brackets, open brackets or parentheses: ( )</li>
    <li>Square brackets, closed brackets or box brackets: [ ]</li>
    <li>curly brackets, squiggly brackets, swirly brackets, braces, or chicken lips: { }</li>
</ul>
<br>

> <h2><strong>Let's Understand the approach first :-</strong></h2>

Reviewer: ```suggestion
<h2><strong>Let's understand the approach first :-</strong></h2>
```

<p>

> <h3>We have to determine whether the given input of Brackets are valid or not.</h3>

Reviewer: ```suggestion
<h3>We have to determine whether the given input of brackets is valid or not.</h3>
```

<ul>
<br>

> <li>To check this we first use stack data structure, which works on a principle of <strong>LIFO (Last in first out) .</strong></li><br>

Reviewer: ```suggestion
<li>To check this we first use stack data structure, which works on the principle of <strong>LIFO (Last in first out) .</strong></li><br>
```

<li>We start iterating over the input string given to us , and in the if clause (statement) whenever we encounter a open bracket i.e. <strong> ( , [ , { </strong> we push that bracket in the stack data structure which we have used.</li>
<br>

<li>And, on the other hand in the else clause (condition).</li>
<br>
<ul>

>     <li>We first always check that our stack should not be empty , because if the stack is empty that implies that the first character we encountered in the string is a close bracket. and if we get this condition we just simply need to return back from the function and no need to check further ,<strong> hence the given input of strings are not Valid Brackets.</strong></li><br>

Reviewer: ```suggestion
    <li>We first always check that our stack should not be empty because if the stack is empty then that implies that the first character we encountered in the string is a close bracket. If we get this condition we just simply need to return back from the function and no need to check further,<strong> hence the given input of strings are not Valid Brackets.</strong></li><br>
```

>     <li>Or, else we pop the value out of the stack and compare it with the character we have encountered with i.e. it's valid combination with the closed brackets. Valid Combinations are as follows :-</li><br>

Reviewer: ```suggestion
    <li>Or, else we pop the value out of the stack and compare it with the character we have encountered with i.e. it's a valid combination with the closed brackets. Valid Combinations are as follows :-</li><br>
```

    <ul>
        <li>Valid Combination for : <strong>(</strong> -> <strong>)</strong></li>
        <li>Valid Combination for : <strong>[</strong> -> <strong>]</strong></li>
        <li>Valid Combination for : <strong>{</strong> -> <strong>}</strong></li>
    </ul>
    <br>
    <li>If we encounter any, other combinations we just simply return back from the function and no need to check further ,<strong> hence the given input of strings are not Valid Brackets.</strong></li><br>
    <li>At last if we came out of the loop , we just need to check one condition whether our stack data structure is empty or not.</li><br>
    <ul> 

>         <li>If it is not empty than it means all of the Brackets are not matched and are not poped out completely. and,<strong> hence the given input of strings are not Valid Brackets.</strong> </li>

Reviewer: ```suggestion
        <li>If it is not empty then it means all of the Brackets are not matched and are not popped out completely and,<strong> hence the given input of strings are not Valid Brackets.</strong> </li>
```

>         <li>Else, if the stack is empty than it means all of the brackets are matched properly and poped out completely. and,<strong> hence at last we can say that the given input of strings are Valid Brackets.</strong> </li>

Reviewer: ```suggestion
        <li>Else, if the stack is empty then it means all of the brackets have matched properly and popped out completely and,<strong> hence, at last, we can say that the given input of strings are Valid Brackets.</strong> </li>
```

    <ul>

        <li>Valid Combination for : <strong>(</strong> -> <strong>)</strong></li>
        <li>Valid Combination for : <strong>[</strong> -> <strong>]</strong></li>
        <li>Valid Combination for : <strong>{</strong> -> <strong>}</strong></li>
    </ul>
    <br>
    <li>If we encounter any, other combinations we just simply return back from the function and no need to check further ,<strong> hence the given input of strings are not Valid Brackets.</strong></li><br>
    <li>At last if we came out of the loop , we just need to check one condition whether our stack data structure is empty or not.</li><br>
    <ul> 
        <li>If it is not empty than it means all of the Brackets are not matched and are not poped out completely. and,<strong> hence the given input of strings are not Valid Brackets.</strong> </li>
        <li>Else, if the stack is empty than it means all of the brackets are matched properly and poped out completely. and,<strong> hence at last we can say that the given input of strings are Valid Brackets.</strong> </li>

>     </ul>

Reviewer 2: Looks like having some syntax error, Once preview and check it.

Author: done sir!
